perception/sandbox/eyehandcal/src/eyehandcal/calibrator.py

```diff
@@ -0,0 +1,118 @@
+
+
+def solve_stage1(proj_func, obs_data_std, K):
+    # stage 1 - assuming marker is attach to EE origin, solve camera pose first
+    if proj_func==hand_marker_proj_world_camera:
+        p3d = torch.stack([p[1] for p in obs_data_std]).detach().numpy()
+    elif proj_func==world_marker_proj_hand_camera:
+        p3d = torch.stack([rotmat(-p[2]).matmul(-p[1]) for p in obs_data_std]).detach().numpy()
+
+    p2d = torch.stack([p[0] for p in obs_data_std]).detach().numpy()
+    retval, rvec, tvec = cv2.solvePnP(p3d, p2d, K.numpy(), distCoeffs=None, flags=cv2.SOLVEPNP_SQPNP)
+    rvec_cam = torch.tensor(-rvec.reshape(-1))
+    tvec_cam = -rotmat(rvec_cam).matmul(torch.tensor(tvec.reshape(-1)))
+    pixel_error = mean_loss(obs_data_std, torch.cat([rvec_cam, tvec_cam, torch.zeros(3)]), K, proj_func).item()
+    print('stage 1 mean pixel error', pixel_error)
+    return rvec_cam, tvec_cam
+
+
+def solve_stage2(proj_func, obs_data_std, K, pixel_tolerance, rvec_cam, tvec_cam):
+    # stage 2 - allow marker to move, joint optimize camera pose and marker
+    max_stage2_retry = 10
+    stage2_retry_count = 0
+    
+    while True :
+        stage2_retry_count += 1
+        if stage2_retry_count > max_stage2_retry:
+            cal_result=CalibrationResult(num_marker_seen=len(obs_data_std),
+                                                    stage2_retry=stage2_retry_count,
+                                                    param=param_star,
+                                                    pixel_error=pixel_error,
+                                                    proj_func=proj_func.__name__)
+            print('Maximum stage2 retry execeeded, bailing out')
+            break
+
+        marker_max_displacement = 0.1 #meter
+        param=torch.cat([rvec_cam, tvec_cam, torch.randn(3)*marker_max_displacement]).clone().detach()
+        param.requires_grad=True
+        L = lambda param: mean_loss(obs_data_std, param, K, proj_func)
+        try:
+            param_star=find_parameter(param, L)
+        except Exception as e:
+            print(e)
+            continue
+
+        pixel_error = L(param_star).item()
+        print('stage 2 mean pixel error', pixel_error)
+        if pixel_error > pixel_tolerance:
+            print(f"Try again {stage2_retry_count}/{max_stage2_retry} because of poor solution {pixel_error} > {pixel_tolerance}")
+        else:
+            print(f"Good solution {pixel_error} <= {pixel_tolerance}")
+            cal_result=CalibrationResult(num_marker_seen=len(obs_data_std),
+                                                    stage2_retry=stage2_retry_count,
+                                                    param=param_star,
+                                                    pixel_error=pixel_error,
+                                                    proj_func=proj_func.__name__)
+            break
+
+    return cal_result
+
+
+def solveEyeHandCalibration(corner_data, proj_func, pixel_tolerance):
```

Comment:
I think Python suggests snake over camel case for functions
```suggestion
def solve_eye_hand_calibration(corner_data, proj_func, pixel_tolerance):
```